debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -402,6 +402,16 @@ public void read(ReplicationMessageProcessor processor) throws SQLException, Int
             public boolean readPending(ReplicationMessageProcessor processor) throws SQLException, InterruptedException {
                 ByteBuffer read = stream.readPending();
                 final long lastReceiveLsn = stream.getLastReceiveLSN().asLong();
+                // Conflict between DBZ-1824 and DBZ-800
+                // PostgreSQL replication stream increases last received LSN even if no message arrives
+                // Then if data message arrives it will have the same LSN as the updated one.
+                // For wal2json it means that the transaction in progress will contain last_processed_lsn to the same value
+                // as transaction LSN and upon restart the incomplete transaction will be skipped instead of being replayed
+                // Also there is an open question whether this can lead to premature LSN flushing as if empty message and data
```

Comment:
That sounds a bit concerning. Are you planning to follow up on this? Anything we should put into the docs?